### PR TITLE
feat(lapis2): stream data from SILO

### LINKED ISSUE #744: Stream sequences
As a counterpart of https://github.com/GenSpectrum/LAPIS-SILO/issues/112:

Stream responses of the sequences endpoints (currently the response is loaded into memory before streaming the result).

### PR DESCRIPTION
**Pull request by fengelniederhammer**
resolves #744

## PR Checklist
- ~~[ ] All necessary documentation has been adapted.~~
- [x] The implemented feature is covered by an appropriate test.

## Summary

The streaming still happens in the same thread as the request handling (by not using `StreamingResponseBody`). We'll have to observe whether this causes any issues in the future. I couldn't properly get using `StreamingResponseBody` work with the way we implemented compression. We have to close the compressions streams [here](https://github.com/GenSpectrum/LAPIS/blob/f6a6184c16127d21ba9acb3d8956a514c89d6999/lapis2/src/main/kotlin/org/genspectrum/lapis/controller/CompressionFilter.kt#L157-L158), but then the lazy stream can't write anymore.

I profiled LAPIS with two datasets.

### TLDR

* Memory consumption when downloading 3.5k sequences goes down from 500 - 900 MB to a 130 - 300 MB.
* Memory consumption when downloading 1M sequence details goes down from 14.5 GB (peak) goes down to a stable usage of ~100MB for the whole download.

### Ebola

I executed this a couple of times with the data of [this](https://github.com/GenSpectrum/LAPIS/tree/ebolaTestData) branch and observed the memory consumption:
```bash
curl -X 'GET' 'http://localhost:8090/sample/alignedNucleotideSequences' \
                   -H 'accept: text/x-fasta' > /dev/null
``` 

Memory consumption on main:
grafik

Memory consumption on this branch:
grafik

### The open Covid dataset on the testserver

I connected my local LAPIS to the SILO running on the server and downloaded metadata:
``` 
curl -X 'GET' 'http://localhost:8090/sample/details?dataFormat=csv&limit=1000000'
```

This is the memory consumption on the current main:
grafik

Also the download was incredibly slow. It took __3.5 minutes__ to downlaod 426 MB:
```
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  426M  100  426M    0     0  2054k      0  0:03:32  0:03:32 --:--:-- 97.7M
``` 

This is the same for the current branch:
grafik

The download isn't fast, but still almost twice as fast:
```
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  426M    0  426M    0     0  3462k      0 --:--:--  0:02:06 --:--:--  2618
```

### CODE REVIEW

File: lapis2/src/test/kotlin/org/genspectrum/lapis/controller/SingleSegmentedSequenceControllerTest.kt [hunk 1]
```diff
@@ -107,19 +112,18 @@ class SingleSegmentedSequenceControllerTest(
             )
         } returns returnedValue
 
-        val request = postSample(ALIGNED_NUCLEOTIDE_SEQUENCES_ROUTE)
+        val request2 = postSample(ALIGNED_NUCLEOTIDE_SEQUENCES_ROUTE)
```
Comment (JonasKellerer) [inline lapis2/src/test/kotlin/org/genspectrum/lapis/controller/SingleSegmentedSequenceControllerTest.kt:115]: This renaming seems unintentional.

**Review by JonasKellerer — APPROVED** [review]
LGTM

**Comment by fengelniederhammer** [comment]
Merging despite red docs tests, since this didn't change anything there.
